internal/cli/init.go

```diff
@@ -220,11 +222,19 @@ func (c *InitCommand) validateConfig() bool {
 	defer sg.Wait()
 
 	s := sg.Add("Validating configuration file...")
-	cfg, err := c.initConfig(c.fromProject, false)
+	cfg, err := c.initConfig(c.fromProject)
 	if err != nil {
 		c.stepError(s, initStepConfig, err)
 		return false
 	}
+	if cfg == nil {
+		// This should never happen, because if there is no config, init should have created
+		// it an exited earlier.
```

Comment:
This comment confuses me. We are in the middle of `init` here, so when exactly should the config have been created earlier? In a different `init`? This one? Is this whole `validateConfig` function pointless?
Also small typo that may be contributing to my confusion.
```suggestion
		// This should never happen, because if there is no config, init should have created
		// it and exited earlier.
```

Reply:
Ah yup, my understanding is that if you run `waypoint init` with no config, it'll generate a new config file for you, and then exit (here, I think: https://github.com/hashicorp/waypoint/pull/2760/files#diff-37f93a8cf9c002499294368250c727ee1e6360f84486e987c97514a2b7cf737aR143-R147).

So I don't think you'd be able to make it past that check, but then not find a config file in this stage.